# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/xhr/XMLHttpRequestMainThread.h
# Commit: 13780f5cb2f5
# Full Hash: 13780f5cb2f5d698f35a23f2fafa1593ef7affe9
# Author: Joshua Marshall <jmarshall@mozilla.com>
# Date: 2024-01-28 21:22:06
# Regressor Bug: 1805613
# File Overlap Count: 1
# Description:
#   Bug 1805613 - Add a mutable status to WorkerRefs for logging. r=dom-worker-reviewers,asuth
#   
#   In DEBUG mode, add a settable string to WorkerRefs.  Record what we're doing in
#   the XHR request worker, and log the status if we seem to be hanging (see
#   WorkerPrivate::DumpCrashInformation).
# ==============================================================================

diff -r 1b609eb33a5f -r 13780f5cb2f5 dom/xhr/XMLHttpRequestMainThread.h
--- a/dom/xhr/XMLHttpRequestMainThread.h	Sun Jan 28 13:43:54 2024 +0100
+++ b/dom/xhr/XMLHttpRequestMainThread.h	Sun Jan 28 14:28:44 2024 +0000
@@ -42,6 +42,7 @@
 #include "mozilla/dom/PerformanceStorage.h"
 #include "mozilla/dom/ServiceWorkerDescriptor.h"
 #include "mozilla/dom/URLSearchParams.h"
+#include "mozilla/dom/WorkerRef.h"
 #include "mozilla/dom/XMLHttpRequest.h"
 #include "mozilla/dom/XMLHttpRequestBinding.h"
 #include "mozilla/dom/XMLHttpRequestEventTarget.h"
@@ -455,6 +456,11 @@
 
   void LocalFileToBlobCompleted(BlobImpl* aBlobImpl);
 
+#ifdef DEBUG
+  // For logging when there's trouble
+  RefPtr<ThreadSafeWorkerRef> mTSWorkerRef = nullptr;
+#endif
+
  protected:
   nsresult DetectCharset();
   nsresult AppendToResponseText(Span<const uint8_t> aBuffer,